src/core/lib/surface/completion_queue.c

```diff
@@ -961,7 +960,7 @@ static void cq_finish_shutdown_next(grpc_exec_ctx *exec_ctx,
   cq_next_data *cqd = (cq_next_data *)DATA_FROM_CQ(cq);
 
   GPR_ASSERT(cqd->shutdown_called);
-  GPR_ASSERT(gpr_atm_no_barrier_load(&cqd->pending_events) == 0);
+  GPR_ASSERT(gpr_atm_acq_load(&cqd->pending_events) == 0);
```

Comment:
What is the purpose here? An assertion is an internal invariant and should be fine even if it didn't exist, so not sure what the ordering is providing.

Reply:
Sure - No barrier does seem fine here. I will change.